src/cpp/communication/CopyPayloadPool.cpp

```diff
@@ -31,41 +31,37 @@ bool CopyPayloadPool::release_payload(
         if (release_payload(cache_change.serializedPayload))
         {
             cache_change.payload_owner(nullptr);
+            return true;
         }
         else
         {
             return false;
         }
     }
-    return false;
+    else
+    {
+        return false;
+    }
 }
 
 bool CopyPayloadPool::get_payload(
         uint32_t size,
         Payload& payload)
 {
-    payload.reserve(size);
+    reserve_(size, payload);
     payload.length = size;
 
-    if (size > 0)
-    {
-        // Set that a payload has been reserved
-        add_reserved_payload_();
-    }
-    else
-    {
-        // Set that a payload has been released
-        add_release_payload_();
-    }
```

Comment:
Should we keep this just in case? Although should never happen.

Reply:
This was my fault, as I did not realize that method `empty` actually release a Payload. (How could I? The name of the method is so descriptive, and it so well commented...).
Thus, it should never call reserve with size 0 (actually this is a good check to add).